ssh/keytypes.pyx

```diff
@@ -31,115 +30,131 @@ cdef class KeyType:
         return self._type
 
     def __str__(self):
-        cdef bytes _type
         cdef const_char *c_type
-        with nogil:
-            c_type = ssh_key_type_to_char(self._type)
-        _type = c_type
-        return to_str(_type)
+        c_type = ssh_key_type_to_char(self._type)
+        if c_type:
```

Comment:
Safer as `if c_type is not NULL`

Reply:
Certainly a little clearer considering Cython's mixed type system; done